File: app/src/main/java/mobi/klimaszewski/demo/multiplatform/MainViewModel.kt
package mobi.klimaszewski.demo.multiplatform

import androidx.lifecycle.ViewModel
import kotlinx.coroutines.flow.MutableStateFlow
import kotlinx.coroutines.flow.StateFlow
import mobi.klimaszewski.demo.common.GetParticipantsUseCase

class MainViewModel: ViewModel() {

    private val getParticipantsUseCase = GetParticipantsUseCase()
    private val _participants = MutableStateFlow(emptyList<String>())
    val participants : StateFlow<List<String>> = _participants

    init {
        _participants.value = getParticipantsUseCase.invoke()
    }
}

File: ios/Shared/MainViewModel.swift
//
//  MainViewModel.swift
//  Multiplatform-ios
//
//  Created by Szymon Klimaszewski on 29/03/2022.
//

import Foundation
import common

class MainViewModel : ObservableObject {
    
    private let getParticipantsUseCase = GetParticipantsUseCase()
    
    @Published var participants : [ParticipantModel]
    
    init() {
        participants = getParticipantsUseCase.invoke().map({ name in
            ParticipantModel(name: name)
        })
    }
    
    struct ParticipantModel : Identifiable, Hashable {
        var name: String
        var id = UUID()
    }
}
